lib/ansible/modules/system/timezone.py

```diff
@@ -353,8 +353,20 @@ def __init__(self, module):
             #   self.allow_no_file['name'] = False <- this is default behavior
             self.conf_files['name'] = '/etc/sysconfig/clock'
             self.conf_files['hwclock'] = '/etc/sysconfig/clock'
-            self.regexps['name'] = re.compile(r'^ZONE\s*=\s*"?([^"\s]+)"?', re.MULTILINE)
-            self.tzline_format = 'ZONE="%s"\n'
+            # The key for timezone might be `ZONE` or `TIMEZONE`
+            # (the former is used in RHEL/CentOS and the latter is used in SUSE linux).
+            # So check the content of /etc/sysconfig/clock and decide which key to use.
+            file = open(self.conf_files['name'], mode='r')
```

Comment:
Could you please replace this `with `

Reply:
Oh, I didn't know that Ansible abandoned python 2.4...cool!

Reply:
Modified: 0d5f2ac